frontend/src/main/scala/bloop/dap/DebuggeeRunner.scala

```diff
@@ -0,0 +1,51 @@
+package bloop.dap
+
+import bloop.data.{Platform, Project}
+import bloop.engine.State
+import bloop.engine.tasks.{RunMode, Tasks}
+import bloop.exec.JavaEnv
+import ch.epfl.scala.bsp.ScalaMainClass
+import monix.eval.Task
+
+trait DebuggeeRunner {
+  def run(logger: DebugSessionLogger): Task[Unit]
+}
+
+private final class MainClassDebugAdapter(
+    project: Project,
+    mainClass: ScalaMainClass,
+    env: JavaEnv,
+    state0: State
+) extends DebuggeeRunner {
+  def run(logger: DebugSessionLogger): Task[Unit] = {
+    val stateForDebug = state0.copy(logger = logger)
+    val workingDir = state0.commonOptions.workingPath
+    val runState = Tasks.runJVM(
+      stateForDebug,
+      project,
+      env,
+      workingDir,
+      mainClass.`class`,
+      mainClass.arguments.toArray,
+      skipJargs = false,
+      RunMode.Debug
+    )
+
+    runState.map(_ => ())
+  }
+}
+
+object DebuggeeRunner {
+  def forMainClass(
+      projects: Seq[Project],
+      mainClass: ScalaMainClass,
+      state: State
+  ): Either[String, DebuggeeRunner] = {
+    val project = projects.head
+    project.platform match {
+      case jvm: Platform.Jvm =>
```

Comment:
Here, we should also be checking if the associated java home to the project is a JDK or not. Most of the times, there is no associated java home, but if there is bloop should be respecting it and using it to fork. There have been several people in Gitter that have complained this is not working correctly, I'm planning to look into it in another PR, but we should nonetheless check those java homes here.

Reply:
>  the JVM on which we run the main class should not, as long as it implements the JDI.
  
It'd be good if you can verify it, but I think you do need to have a JDK in both the host VM and the forked VM for debugging to work.

Reply:
I have created a simple java class and then run it with JDI enabled on both JDK and JRE:
It prints the compiler it gets by using  `javax.tools.ToolProvider.getSystemJavaCompiler()`

```
$ java -version
openjdk version "1.8.0_222"
OpenJDK Runtime Environment (build 1.8.0_222-b05)
OpenJDK 64-Bit Server VM (build 25.222-b05, mixed mode)

$ java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,quiet=n Foo 
Listening for transport dt_socket at address: 41419
Compiler: com.sun.tools.javac.api.JavacTool@1175e2db
```

```
$ /usr/lib/jvm/java-9-jre/jre/bin/java -version
java version "9.0.4"
Java(TM) SE Runtime Environment (build 9.0.4+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.4+11, mixed mode)

$ /usr/lib/jvm/java-9-jre/jre/bin/java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,quiet=n Foo 
Listening for transport dt_socket at address: 43201
Compiler: null
```
Since both of those are listening for `transport dt_socket...`, it looks like we don't need JDK on the debuggee VM, just the host.

Reply:
Good catch :+1: I'm happy we don't need to check the java homes we use to run/test.